cmd/bundle/summary.go

```diff
@@ -0,0 +1,67 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/spf13/cobra"
+)
+
+func newSummaryCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "summary",
+		Short: "Describe the bundle resources and their deployment states",
+
+		PreRunE: ConfigureBundleWithVariables,
+
+		// This command is currently intended for the Databricks VSCode extension only
+		Hidden: true,
+	}
+
+	var forcePull bool
+	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Skip local cache and load the state from the remote workspace")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		b := bundle.Get(cmd.Context())
+
+		err := bundle.Apply(cmd.Context(), b, phases.Initialize())
+		if err != nil {
+			return err
+		}
+
+		cacheDir, err := terraform.Dir(cmd.Context(), b)
+		if err != nil {
+			return err
+		}
+		_, err = os.Stat(filepath.Join(cacheDir, terraform.TerraformStateFileName))
+		noCache := errors.Is(err, os.ErrNotExist)
+
+		if forcePull || noCache {
+			err = bundle.Apply(cmd.Context(), b, terraform.StatePull())
+			if err != nil {
+				return err
+			}
+		}
+
+		err = bundle.Apply(cmd.Context(), b, terraform.Load())
+		if err != nil {
+			return err
+		}
+
+		buf, err := json.MarshalIndent(b.Config, "", "  ")
+		if err != nil {
+			return err
+		}
+
+		cmd.OutOrStdout().Write(buf)
```

Comment:
Please add a check on the output type. There is not enough precedence of doing this, but I'd like to make sure all commands have a JSON output mode as well as a text (human readable, on by default) output mode. The run command implements this like so:

```go
			switch root.OutputType(cmd) {
			case flags.OutputText:
				resultString, err := output.String()
				if err != nil {
					return err
				}
				cmd.OutOrStdout().Write([]byte(resultString))
			case flags.OutputJSON:
				b, err := json.MarshalIndent(output, "", "  ")
				if err != nil {
					return err
				}
				cmd.OutOrStdout().Write(b)
			default:
				return fmt.Errorf("unknown output type %s", root.OutputType(cmd))
			}
```

Could you add the same here with the text output being tbd? Then callers will need to specify `--output json`.